Updates to better support TypeScript (#2)
zeroc-ice/gulp-ice-builder: README.md

### jsbundle `Boolean`

Create a JavaScript bundle for each `js:module`, the bundle contains the JavaScript generated
code for all the Slice compilation units that belong to a given `js:module`, an extra bundled
named `generated.js` is generated containing all the JavaScript generated code for Slice compilation
units that doesn't belong to any `js:module`.

The `jsbundle` option is enabled by default when `args` contains `--typescript` value, it can
be manually enabled and disabled by setting this property to `true` or `false` respectively.

The bundle creation uses [Rollup][3] module bundler and it requires that your Slice definitions
use the [es6 JavaScript module mapping][4] introduced with Ice 3.7.

```js
iceBuilder({
    args: ["--typescript"],
    jsbundle: false});
```

### jsbundleFormat `String`

The output format use by Rollup generated bundled, it correspond to Rollup `--format` option. The accepted
values are `amd`, `cjs`, `es`, `iife` and  `umd`. The default value is `es`.

```js
iceBuilder({
    args: ["--typescript"],
    jsbundleFormat: "cjs"});
```

### jsbundleSourcemap `Boolean`

> Enable or disable the generation of sourcemap files for the generated JavaScript bundle. The default is to

Reviewer: ```suggestion
Enable or disable the generation of source map files for the generated JavaScript bundle. The default is to
```

generate a source map for each generated bundled, it can be disabled by setting this option to `false`.

```js
iceBuilder({
    args: ["--typescript"],
    jsbundleSourcemap: false});
```

### tsbundle `Boolean`

> Create a bundle containing the TypeScript declarations for each `js:module`, an extra

Reviewer: ```suggestion
Create a bundle containing the TypeScript declarations for each `js:module`. An extra
```

> bundled named `generated.d.ts` is generated containing the TypeScript generated declarations

Reviewer: ```suggestion
bundle named `generated.d.ts` is generated containing the TypeScript generated declarations
```

> for Slice compilation units that doesn't belong to any `js:module`.

Reviewer: ```suggestion
for Slice compilation units that do not belong to any `js:module`.
```

```js
iceBuilder({
    args: ["--typescript"],
    tsbundle: false});
```

> The `tsbundle` option is enabled by default when `args` contains `--typescript` value, it can

Reviewer: ```suggestion
The `tsbundle` option is enabled by default when `args` contains `--typescript` value. It can
```

```

For a full list of arguments you can pass to the `slice2js` compiler refer to [slice2js](https://github.com/zeroc-ice/npm-slice2js).

### jsbundle `Boolean`

The creation of JSBundle is enabled by default when `--typescript` argument is used, it can be disabled
by setting `jsbundle` property to `false`. The creation of JavaScript bundles use [Rollup](1) module bundler

> and it requires that your Slice definitions use the [es6 JavaScript module mapping](2).

Reviewer: This link does not work.

Reviewer: Same for the one on the line above.
